### PR TITLE
Decoding u128

### PR DESCRIPTION
**Pull request by Daniel-K-Ivanov**

### CODE REVIEW

File: assembly/__tests__/UInt.spec.ts [hunk 1]
```diff
@@ -179,4 +180,44 @@ describe("UInt64", () => {
...
+        const v5 = UInt128.fromU8a([0xfe, 0xff, 0xff, 0xff]);
+        expect<string>(v5.toString()).toStrictEqual("1073741823");
+
+        const v6 = UInt128.fromU8a([0x03, 0xff, 0xff, 0xff, 0xff]);
+        expect<string>(v6.toString()).toStrictEqual("4294967295");
+
+        const v7 = UInt128.fromU8a([0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff]);
+        expect<string>(v7.toString()).toStrictEqual("1329227995784915872903807060280344575");
+
+        const v8 = UInt128.fromU8a([ 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff, 0x12]);
```
Comment (bakasura980) [inline assembly/__tests__/UInt.spec.ts:217]: Woow, how that is equal to 2180. I mean above v4 is 4 bytes long and it is 16384 which is bigger

File: assembly/utils/Bytes.ts [hunk 2]
```diff
@@ -50,7 +50,22 @@ export class Bytes {
     }
 
     /**
-    * @description Copy u8[] src elements in Uint8Array dst at provided position. 
+     * Copies one Array into another by a given start position
+     * @param src - Source array from which we are copying
+     * @param dst - Destination array to which we are copying
+     * @param start - The start index of the Source array that we are copying from
+     */
+    static copyFrom<T>(src: T[], dst: Array<T>, start: i32 = 0): void {
```
Comment (bakasura980) [inline assembly/utils/Bytes.ts]: Where do we use this function

File: assembly/utils/Bytes.ts [hunk 3]
```diff
@@ -96,6 +111,30 @@ export class Bytes {
...
+
+        const mode = input[0] & 0x03;
+        if (i32(mode) <= BIT_LENGTH.INT_16) {
+            return [u64(Bytes.decodeSmallInt(input, mode).value), 0];
+        }
+
+        const valueLE = input.slice(1);
+        const buffer = valueLE.dataStart;
+        const lo = load<u64>(buffer, 0);
+        const hi = valueLE.length <= 8 ? 0 : load<u64>(buffer, 1 * sizeof<u64>());
```
Comment (bakasura980) [inline assembly/utils/Bytes.ts]: Does this work with 10 bytes number?

File: assembly/UInt/UInt128.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { u128 } from "as-bignum";
+import { Bytes } from "../utils/Bytes";
+
+/** Representation for a UInt128 value in the system. */
+export class UInt128 extends u128 {
+
+    /** Instantiates new UInt128 from u8[] SCALE encoded bytes */
+    static fromU8a(value: u8[]): UInt128 {
+        const result = Bytes.decodeU128(value);
```
Comment (bakasura980) [inline assembly/UInt/UInt128.ts]: decodeU128 is the actual implementation of UInt128. Why we get it from utils?

**Review by bakasura980 — CHANGES_REQUESTED** [review]

File: assembly/__tests__/UInt.spec.ts (hunk 1 above)
Comment (Daniel-K-Ivanov) [reply assembly/__tests__/UInt.spec.ts:217]: the first byte indicates that the mode is a two-byte mode meaning that the value is only the contained in the first 2 bytes, the rest of them are ignored

File: assembly/utils/Bytes.ts (hunk 2 above)
Comment (Daniel-K-Ivanov) [reply assembly/utils/Bytes.ts]: it is unused, I will delete it

File: assembly/utils/Bytes.ts (hunk 3 above)
Comment (Daniel-K-Ivanov) [reply assembly/utils/Bytes.ts]: yes it does. I will add test

**Review by bakasura980 — APPROVED** [review]

**Pull request merged**